compiler/rustc_session/src/session.rs

```diff
@@ -658,6 +658,11 @@ impl Session {
 
     /// Calculates the flavor of LTO to use for this compilation.
     pub fn lto(&self) -> config::Lto {
+        // if autodiff is enabled, use fat lto
```

Comment:
Can you change this to explain why we do it? Something along the lines of

// Autodiff currently requires fat-lto to have access to the llvm-ir of all (indirectly) used functions and types. fat-lto is the easiest solution to this requirement, but quite expensive. FIXME(autodiff): Make autodiff also work with embed-bc instead of fat-lto.

Otherwise lgtm. Sorry for taking so long, I was traveling and wanted to test it locally.

Reply:
Also, while already at it, can you please remove the now unneeded (and already unused) error related to it? There's also a diagnostics in a .ftl file.
``
compiler/rustc_codegen_ssa/src/errors.rs
42:pub struct AutodiffWithoutLto;
```